src/index.tsx

```diff
@@ -237,21 +240,13 @@ export class App extends React.Component<any, AppState> {
     return true;
   }
 
-  public async componentDidMount() {
-    document.addEventListener("copy", this.onCopy);
-    document.addEventListener("paste", this.onPaste);
-    document.addEventListener("cut", this.onCut);
-
-    document.addEventListener("keydown", this.onKeyDown, false);
-    document.addEventListener("mousemove", this.updateCurrentCursorPosition);
-    window.addEventListener("resize", this.onResize, false);
-    window.addEventListener("unload", this.onUnload, false);
-
+  private async setAppState(id: string | null) {
```

Comment:
The name suggests it sets the `AppState`, which it does not. It restores *scene* from `id` or `localStorage`. I'd call this `restoreScene`.